core/src/ast/mod.rs

```diff
@@ -42,7 +42,59 @@ pub struct ForeignKey {
 #[derive(PartialEq, Debug, Clone, Eq, Hash, Serialize, Deserialize, Display)]
 pub enum ReferentialAction {
     #[strum(to_string = "NO ACTION")]
+    /// A NO ACTION constraint specifies that when a referenced row is deleted,
+    /// no action should be taken.
     NoAction,
+    #[strum(to_string = "CASCADE")]
+    /// A CASCADE constraint specifies that when a referenced row is deleted,
+    /// row(s) that reference it should also be deleted.
+    Cascade,
+    #[strum(to_string = "SET NULL")]
+    /// A SET NULL constraint specifies that when a referenced row is deleted,
+    /// row(s) that reference it should have their referencing column(s) set to NULL.
+    SetNull,
+    #[strum(to_string = "SET DEFAULT")]
+    /// A SET DEFAULT constraint specifies that when a referenced row is deleted,
+    /// row(s) that reference it should have their referencing column(s) set to the column's default value.
+    SetDefault,
+    #[strum(to_string = "RESTRICT")]
+    /// A RESTRICT constraint specifies that when a referenced row is attempted to be deleted,
+    /// the deletion should be restricted i.e. prevented.
+    Restrict,
+}
+
+impl ReferentialAction {
+    /// Returns whether the referential action is restricting the deletion of the referenced row.
+    pub fn is_restrict(&self) -> bool {
+        matches!(self, ReferentialAction::Restrict) || matches!(self, ReferentialAction::NoAction)
+    }
+
+    /// Returns whether the referential action is cascading the deletion of the referenced row.
+    pub fn is_cascade(&self) -> bool {
+        matches!(self, ReferentialAction::Cascade)
+    }
+
+    /// Returns whether the referential action is setting the referencing column to NULL.
+    pub fn is_set_null(&self) -> bool {
+        matches!(self, ReferentialAction::SetNull)
+    }
+
+    /// Returns whether the referential action is setting the referencing column to the column's default value.
+    pub fn is_set_default(&self) -> bool {
+        matches!(self, ReferentialAction::SetDefault)
+    }
+}
```

Comment:
Seems like these methods were not used at all.

Reply:
I will remove those

Reply:
Done!